README.md
# Analysis-of-COVID-19-rumors
Get COVID-19 rumors and analyze them


data/news.txt
title	date	sourse	content	link

唐山一医院科室因哈尔滨来的患者发烧致全科隔离等待核酸检测？谣言	2020-05-22	河北日报微信公号	[　　5月21日，华北理工大学附属医院发布声明：近日，网传煤医心内科因哈尔滨来的患者发烧致全科隔离等待核酸检测一事纯属谣传。目前，华北理工大学附属医院心内科全体医务人员正在紧张有序开展正常诊疗，就诊秩序良好。疫情以来，华北理工大学附属医院始终紧抓疫情防控，严格就医流程，关注患者健康，以病人为中心营造安全的诊疗环境。目前，华北理工大学附属医院各门诊及所有住院病区均正常开诊，请大家放心就诊。]	http://www.piyao.org.cn/2020-05/22/c_1210629519.htm

丹阳六中出现学生无症状感染者？谣言	2020-04-17	丹阳发布	[　　15日，丹阳发布对关于“丹阳市出现无症状感染者”传言的情况进行了通报。我市教育、公安两部门迅速开展调查，依法查处相关谣言案件，查处结果如下：, <p align="center"><img align="center" border="0" id="{E7981A0D-32FF-4960-B7B1-B00DD3CE0D2E}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901498981n.jpg" style="HEIGHT: 891px; WIDTH: 456px" title=""/>, <p align="center"><img border="0" id="{926C6475-9805-413B-8A9E-3153FF318A57}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901608041n.jpg" style="HEIGHT: 607px; WIDTH: 455px" title=""/>]	http://www.piyao.org.cn/2020-04/17/c_1210562170.htm

酒精消毒，75%浓度效果最佳！	2020-02-12	人民网	[　　医用酒精消毒属于化学消毒法，使用时要注意：75%浓度的效果最佳。特别提醒，不要把酒精大量喷洒在婴幼儿和成人身体上，一方面防止酒精过敏，另一方面也要注意使用时不可靠近火源和热源，防止火灾。学名乙醇的酒精，在常温常压下就是一种易燃、易挥发的无色透明液体，闪点仅13℃，扩散后蒸汽与空气形成爆炸性混合物，遇明火、高热能引起爆炸燃烧。给电器表面消毒，应先关闭电源，待电器冷却后再进行，如用酒精擦拭厨房灶台，要先关闭火源，以免导致爆燃。每次取用酒精后必须立即将容器上盖封闭。家庭不要大量囤积酒精！, 　　如果酒精浓度过高（例如 90%），需稀释至75%左右使用，否则无法有效杀灭病毒。, ]	http://www.piyao.org.cn/2020-02/12/c_1210470785.htm

30万湖北籍境外旅客打包回上海机场？谣言	2020-02-12	上海网络辟谣	[　　近日，有消息称：“杨州旅行社刚接到重要通知，现在境外所有的湖北籍三十万游客，全部打包返回中国上海浦东机场，交由上海市管制。”, <p align="center"><img align="center" border="0" id="{D11C5E4C-C23C-49A0-9341-E830C8AD8670}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210470752_15814409187121n.jpg" style="HEIGHT: 351px; WIDTH: 478px" title=""/>, 　　搜索发现，上海、湖北都没有找到以“杨州”为名的地点，两地也没有以“杨州”为名的旅行社。另一方面，记者以旅客身份联系到上海浦东机场工作人员，对方表示暂时没有接到相关通知，建议以官方通知为准。事实上，目前正在境外旅游的武汉游客，大多正为如何返回而苦恼。]	http://www.piyao.org.cn/2020-02/12/c_1210470752.htm

大庆驰援武汉医疗队技术不过硬，被遣送？不实	2020-03-01	中国经济网	[　　近日，一微信名为“王*涵”的网民在大庆一微信群（群内成员400余人）内发布信息称：“我刚才听到个好玩的事，咱们大庆派到武汉支援的人员都给遣送回来了，说咱们技术不过硬，不是外科的就是眼科的牙科的，去了连防护服都穿不明白，人家还得操心照顾他们，都给遣回来了，现在在大庆隔离呢”。据大庆网警巡查执法消息，经与大庆市卫健委核实，疫情发生以来，大庆市先后组织两批医疗队员，包括重症医学科、呼吸内科、中西医结合科等共计13名骨干医生、20名优秀护士，奔赴湖北省武汉市及孝感市进行医疗救援。其中，三名医生到达武汉后第一时间进入武汉方舱医院，其余医护人员到达孝感后第一时间至安陆市人民医院开展医疗救助工作。据悉，诊疗中，大庆医疗队采取中西医结合诊疗方式，对病人进行精准化诊治，“暖心”式心理关爱取得良好效果，捷报频传。2月26日，35名患者治愈出院，是大庆医疗队进驻安陆市人民医院以来所管患者出院人数最多的一次。2月29日，大庆市公安局龙岗分局将违法行为人王某 (女,40岁， 大庆人)抓获。经审讯，王某对其在微信群内造谣抹黑医护人员的违法行为供认不讳。根据《治安管理处罚法》第二十五条第一款之规定，王某被大庆市公安局龙岗分局行政拘留7日并处罚款500元 。（记者朱国旺 郭文培）]	http://www.piyao.org.cn/2020-03/01/c_1210496485.htm
title	date	sourse	content	link
唐山一医院科室因哈尔滨来的患者发烧致全科隔离等待核酸检测？谣言	2020-05-22	河北日报微信公号	[　　5月21日，华北理工大学附属医院发布声明：近日，网传煤医心内科因哈尔滨来的患者发烧致全科隔离等待核酸检测一事纯属谣传。目前，华北理工大学附属医院心内科全体医务人员正在紧张有序开展正常诊疗，就诊秩序良好。疫情以来，华北理工大学附属医院始终紧抓疫情防控，严格就医流程，关注患者健康，以病人为中心营造安全的诊疗环境。目前，华北理工大学附属医院各门诊及所有住院病区均正常开诊，请大家放心就诊。]	http://www.piyao.org.cn/2020-05/22/c_1210629519.htm
丹阳六中出现学生无症状感染者？谣言	2020-04-17	丹阳发布	[　　15日，丹阳发布对关于“丹阳市出现无症状感染者”传言的情况进行了通报。我市教育、公安两部门迅速开展调查，依法查处相关谣言案件，查处结果如下：, <p align="center"><img align="center" border="0" id="{E7981A0D-32FF-4960-B7B1-B00DD3CE0D2E}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901498981n.jpg" style="HEIGHT: 891px; WIDTH: 456px" title=""/>, <p align="center"><img border="0" id="{926C6475-9805-413B-8A9E-3153FF318A57}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901608041n.jpg" style="HEIGHT: 607px; WIDTH: 455px" title=""/>]	http://www.piyao.org.cn/2020-04/17/c_1210562170.htm
酒精消毒，75%浓度效果最佳！	2020-02-12	人民网	[　　医用酒精消毒属于化学消毒法，使用时要注意：75%浓度的效果最佳。特别提醒，不要把酒精大量喷洒在婴幼儿和成人身体上，一方面防止酒精过敏，另一方面也要注意使用时不可靠近火源和热源，防止火灾。学名乙醇的酒精，在常温常压下就是一种易燃、易挥发的无色透明液体，闪点仅13℃，扩散后蒸汽与空气形成爆炸性混合物，遇明火、高热能引起爆炸燃烧。给电器表面消毒，应先关闭电源，待电器冷却后再进行，如用酒精擦拭厨房灶台，要先关闭火源，以免导致爆燃。每次取用酒精后必须立即将容器上盖封闭。家庭不要大量囤积酒精！, 　　如果酒精浓度过高（例如 90%），需稀释至75%左右使用，否则无法有效杀灭病毒。, ]	http://www.piyao.org.cn/2020-02/12/c_1210470785.htm
30万湖北籍境外旅客打包回上海机场？谣言	2020-02-12	上海网络辟谣	[　　近日，有消息称：“杨州旅行社刚接到重要通知，现在境外所有的湖北籍三十万游客，全部打包返回中国上海浦东机场，交由上海市管制。”, <p align="center"><img align="center" border="0" id="{D11C5E4C-C23C-49A0-9341-E830C8AD8670}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210470752_15814409187121n.jpg" style="HEIGHT: 351px; WIDTH: 478px" title=""/>, 　　搜索发现，上海、湖北都没有找到以“杨州”为名的地点，两地也没有以“杨州”为名的旅行社。另一方面，记者以旅客身份联系到上海浦东机场工作人员，对方表示暂时没有接到相关通知，建议以官方通知为准。事实上，目前正在境外旅游的武汉游客，大多正为如何返回而苦恼。]	http://www.piyao.org.cn/2020-02/12/c_1210470752.htm
大庆驰援武汉医疗队技术不过硬，被遣送？不实	2020-03-01	中国经济网	[　　近日，一微信名为“王*涵”的网民在大庆一微信群（群内成员400余人）内发布信息称：“我刚才听到个好玩的事，咱们大庆派到武汉支援的人员都给遣送回来了，说咱们技术不过硬，不是外科的就是眼科的牙科的，去了连防护服都穿不明白，人家还得操心照顾他们，都给遣回来了，现在在大庆隔离呢”。据大庆网警巡查执法消息，经与大庆市卫健委核实，疫情发生以来，大庆市先后组织两批医疗队员，包括重症医学科、呼吸内科、中西医结合科等共计13名骨干医生、20名优秀护士，奔赴湖北省武汉市及孝感市进行医疗救援。其中，三名医生到达武汉后第一时间进入武汉方舱医院，其余医护人员到达孝感后第一时间至安陆市人民医院开展医疗救助工作。据悉，诊疗中，大庆医疗队采取中西医结合诊疗方式，对病人进行精准化诊治，“暖心”式心理关爱取得良好效果，捷报频传。2月26日，35名患者治愈出院，是大庆医疗队进驻安陆市人民医院以来所管患者出院人数最多的一次。2月29日，大庆市公安局龙岗分局将违法行为人王某 (女,40岁， 大庆人)抓获。经审讯，王某对其在微信群内造谣抹黑医护人员的违法行为供认不讳。根据《治安管理处罚法》第二十五条第一款之规定，王某被大庆市公安局龙岗分局行政拘留7日并处罚款500元 。（记者朱国旺 郭文培）]	http://www.piyao.org.cn/2020-03/01/c_1210496485.htmtitle	date	sourse	content	link
唐山一医院科室因哈尔滨来的患者发烧致全科隔离等待核酸检测？谣言	2020-05-22	河北日报微信公号	[　　5月21日，华北理工大学附属医院发布声明：近日，网传煤医心内科因哈尔滨来的患者发烧致全科隔离等待核酸检测一事纯属谣传。目前，华北理工大学附属医院心内科全体医务人员正在紧张有序开展正常诊疗，就诊秩序良好。疫情以来，华北理工大学附属医院始终紧抓疫情防控，严格就医流程，关注患者健康，以病人为中心营造安全的诊疗环境。目前，华北理工大学附属医院各门诊及所有住院病区均正常开诊，请大家放心就诊。]	http://www.piyao.org.cn/2020-05/22/c_1210629519.htm
丹阳六中出现学生无症状感染者？谣言	2020-04-17	丹阳发布	[　　15日，丹阳发布对关于“丹阳市出现无症状感染者”传言的情况进行了通报。我市教育、公安两部门迅速开展调查，依法查处相关谣言案件，查处结果如下：, <p align="center"><img align="center" border="0" id="{E7981A0D-32FF-4960-B7B1-B00DD3CE0D2E}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901498981n.jpg" style="HEIGHT: 891px; WIDTH: 456px" title=""/>, <p align="center"><img border="0" id="{926C6475-9805-413B-8A9E-3153FF318A57}" sourcedescription="编辑提供的本地文件" sourcename="本地文件" src="1210562170_15870901608041n.jpg" style="HEIGHT: 607px; WIDTH: 455px" title=""/>]	http://www.piyao.org.cn/2020-04/17/c_1210562170.htm

Code/get_inmongo.py
#!/usr/bin/env python 
# -*- coding:utf-8 -*-

#   LYS
#   2020/6/26 21:19
#   get_inmongo.py

from urllib import request
from bs4 import BeautifulSoup
import re
import os
import bs4
import time
import requests
import re
import pymongo


def get_HTMLText(url):
    headers = {
        'User-Agent': 'user-agent: Mozilla/5.0 (Windows NT 10.0;Win64;x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36',
        'Cookie': 'ncbi_sid=47EF8993E4BE3193_0587SID; _ga=GA1.2.780797648.1582031800; pmc.article.report=; books.article.report=; _gid=GA1.2.288894097.1585885708; QSI_HistorySession=https%3A%2F%2Fwww.ncbi.nlm.nih.gov%2F~1585922167314; WebEnv=1zmWDqnmAX_v2YHx4rlkrq3Yhg3ZCIGss-UrAFg-SZPikri1ywXU5zjN2MwUgcrtonBWnvUs1EJyzzVzB2wC14pFkl6eh-708Vl%4047EF8993E4BE3193_0587SID; ncbi_pinger=N4IgDgTgpgbg+mAFgSwCYgFwgCIAYBC2ArAMIDMRAHGdmQCwCcAbEbm7mbgwOxEBiDWvzoA6AIwiAtnEogANCACuAOwA2AewCGqZVAAeAF0ygATJhABzAI6KoEAJ7yQZc5tWqndcwGcomiADGiE5E5hCKqlAAvIj2qBCaAD6oAEZRYIopklCoid7qioHRJMgGmgbI6soAylDK+RB5galRAAqZALI5iQb2YNH+FQGRiU5iXli+/kFOJrjmeISkFNS0jCzsHFy8AkJ8ohLSsgomYubWtg4YbqoYU4GIGOGRMXEJyWkZWd35hQHFpXKlRqdQaTQCLXaKS6uV6/Sig2QwygoxOZiwAHcsSJlAEUsgcapJDjkIgRBZ1DBZgxzGIGHMnJxabhKPMFGQzlg6Qz2eiQGJcEQXOyXFgDOEoIyJiBxbZGbIsIyaVgiELBTSFHR5lgmDwGJRQprRSBcCITJQRMKQHRpSoNNpdIZPKEsGyQELaawQtKGFqQkxaUwOCFuLSvAomJz+dxuLIAL5xoA'}
    try:
        r = requests.get(url, headers=headers)
        r.raise_for_status()
        r.encoding = r.apparent_encoding
        return r.text
    except:
        return 'error'


def page_parser(url):
    html = get_HTMLText(url)
    soup = BeautifulSoup(html, 'html.parser')
    return soup


def get_connection(url, collection):
    soup = page_parser(url)

    try:
        title = soup.select("div.con_tit > h2")[0].text  # '\r\n电影院等密闭式娱乐休闲场所低中高风险地区均暂不开业\r\n'
        date = soup.select("div.con_tit > p >span")[0].text  # '发布时间：  2020-04-10 14:32:12'
        content = soup.select("div.con_txt > p")[0].text
        # '\u3000\u3000日前，国务院联防联控机制印发通知，落实分区分级防控要求，推进生产生活秩序逐步恢复，进一步做好重点场所、重点单位、重点人群的疫情防控工作。明确了密闭式娱乐、休闲场所低、中、高风险地区均暂不开业，具体要求由各地依据本地疫情形势研究确定。建议低风险地区生活服务类场所在做好室内通风、环境清洁消毒、人员健康监测的前提下正常营业；中、高风险地区生活服务类场所应当限制人员数量，减少人群聚集。（记者 王秉阳）'

        sourse = soup.select("div.con_tit > p")[0].text  # '\r\n来源： 新华社“新华视点”微博发布时间：  2020-04-10 14:32:12\n'
    except:
        pass
    else:
        t = title.strip()
        # print(t)  # title
        s = sourse.strip()
        # print(s)  # sourse

        d = re.findall(r'[0-9-]{10}', date.strip())
        d = d[0]
        # print(d)  # date
        c = content.strip()
        # print(c)  # content

        line = {'title': t, 'date': d, 'sourse': s, 'content': c, 'link': url}
        # print(line)
        collection.insert_one(line)


def page_url(url, url2):
    dict = []
    for x in range(0, 100):
        url3 = url + str(x + 1) + url2 + str(x)
        # print(x)
        html = get_HTMLText(url3)
        soup = BeautifulSoup(html, 'html.parser')
        s = str(soup)
        urls = re.findall('[a-zA-Z0-9/\._]+\.htm', s)

        for k in urls:
            dict.append(k)
    set_url = set(dict)
    # print(dict)
    return set_url


if __name__ == "__main__":
    new = []
    url = 'http://qc.wa.news.cn/nodeart/list?nid=11215616&pgnum='
    url2 = '&cnt=10&attr=63&tp=1&orderby=1&callback=jQuery11240023171643210465165_1593738848999&_=1593738849'
    # 1593738849000
    urls = page_url(url, url2)
    i = 0
    URL = 'http://www.piyao.org.cn/2020-'
    # 连接数据库
    client = pymongo.MongoClient(host='localhost', port=27017)
    db = client['News']
    collection = db['a03']
    for line in urls:
        u = URL + line
        get_connection(u, collection)
